### PR TITLE
Add auth example

### PR DESCRIPTION
**Pull request by felixmosh**
This PR adds an example with passport auth.

### CODE REVIEW

File: examples/with-auth/index.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+// Configure the local strategy for use by Passport.
+//
+// The local strategy require a `verify` function which receives the credentials
+// (`username` and `password`) submitted by the user.  The function must verify
+// that the password is correct and then invoke `cb` with a user object, which
+// will be set at `req.user` in route handlers after authentication.
+passport.use(new LocalStrategy(
+  function (username, password, cb) {
+    if (username === 'bull' && password === 'board') {
```
Comment (joaovbibiano) [inline examples/with-auth/index.js:19]: Should we put it insde the .env file? I think is the best solution at first glance

**Review by joaovbibiano — APPROVED** [review]

File: examples/with-auth/index.js (hunk 1 above)
Comment (felixmosh) [reply examples/with-auth/index.js:19]: it is just an POC example, do you think that it is better to follow the best practices?

Comment (joaovbibiano) [reply examples/with-auth/index.js:19]: Hahah sorry @felixmosh I'm new to reviewing open source projects.

The main idea seems good to me, I think it's simple and I solve our problem.

Comment (felixmosh) [reply examples/with-auth/index.js:19]: I don't think that there is much "special" here, just a simple passport example (which is copied with the comments from passport's example :] )

Comment (joaovbibiano) [reply examples/with-auth/index.js:19]: Ok, good.

Now, we need other Collaborator approve this PR? So, we can finish the implementation? What is the worflow?

**Comment by felixmosh** [comment]
@joaovbibiano can you approve this PR?

File: examples/with-auth/index.js (hunk 1 above)
Comment (felixmosh) [reply examples/with-auth/index.js:19]: Just approve the PR it self, then I will be able to merge it, thanx!

**Review by joaovbibiano — APPROVED** [review]

**Review by joaovbibiano — APPROVED** [review]

**Review by joaovbibiano — APPROVED** [review]
This PR is directly simple, and solves the proposed problem

**Review by vcapretz — APPROVED** [review]

**Pull request merged**